specification/eventgrid/data-plane/Microsoft.EventGrid/preview/2023-06-01-preview/EventGrid.json

```diff
@@ -672,6 +672,146 @@
         "failedLockTokens",
         "succeededLockTokens"
       ]
+    },
+    "EventGridNamespaceMQTTClientCreatedOrUpdatedEventData": {
+      "type": "object",
+      "description": "Event data for Microsoft.EventGrid.MQTTClientCreatedOrUpdated event.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MQTTClientCreatedOrUpdatedEventProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "EventGridNamespaceMQTTClientDeletedEventData": {
+      "type": "object",
+      "description": "Event data for Microsoft.EventGrid.MQTTClientDeleted event.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MQTTClientDeletedEventProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "EventGridNamespaceMQTTClientSessionConnectedEventData": {
+      "type": "object",
+      "description": "Event data for Microsoft.EventGrid.MQTTClientSessionConnected event.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MQTTClientSessionConnectedEventProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "EventGridNamespaceMQTTClientSessionDisconnectedEventData": {
+      "type": "object",
+      "description": "Event data for Microsoft.EventGrid.MQTTClientSessionDisconnected event.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MQTTClientSessionDisconnectedEventProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "MQTTClientCreatedOrUpdatedEventProperties": {
+      "type": "object",
+      "description": "Schema of the Data property of an EventGridEvent for MQTT Client Created Or Updated event.",
+      "properties": {
+        "clientAuthenticationName": {
+          "type": "string",
+          "description": "Unique identifier for the MQTT client that the client presents to the service for authentication. This case-sensitive string can be up to 128 characters long, and supports UTF-8 characters."
+        },
+        "clientName": {
+          "type": "string",
+          "description": "Name of the client resource in the Event Grid namespace."
+        },
+        "namespaceName": {
+          "type": "string",
+          "description": "Name of the Event Grid namespace where the MQTT client was created or updated."
+        },
+        "state": {
```

Comment:
This should be an extensible enum with the known values specified. Here is an example - https://github.com/l0lawrence/azure-rest-api-specs/blob/egevents_restructure/specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json#L497-L507